x/stakeibc/keeper/icqcallbacks_withdrawal_balance.go

```diff
@@ -62,32 +58,15 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "no fee account found for %s", chainId)
 	}
 
-	// Determine the stride commission rate to the relevant portion can be sent to the fee account
-	params := k.GetParams(ctx)
-	strideCommissionInt, err := cast.ToInt64E(params.StrideCommission)
+	// Split the withdrawal amount into a rebate, stride fee, and reinvest portion
+	feeInfo, err := k.CheckForCommunityPoolRebate(ctx, chainId, withdrawalBalanceAmount)
```

Comment:
This is a logic error to call this here unmodified because it will compute and remove the rebateFee again...
The amount being passed in here is `withdrawalBalanceAmount` which measures the DYDX in the withdrawal -- ie, the output of the swap.  The USDC rebate was already sent back before the swap.  This call will set aside some proportion of the DYDX amount now for the rebateFee even though we don't do anything with it.


Concrete numerical example: CP has all of the stDYDX tokens and rebate of 50% on them. Stride fee is 10%. We get 100 USDC reward, compute at that point rebateFee = 5 USDC, strideFee = 5 USDC, reinvest = 90 USDC all correct so far.  Then we send off 95 USDC to be swapped and it comes back as 95 DYDX assuming a 1:1 price for simplicity.  Now you get to this call back and withdrawalBalanceAmount is 95 representing the 95 DYDX post swap.  And it tries to compute rebateFee = 4.75 DYDX, stride fee = 4.75 DYDX and reinvest = 85.5 DYDX when it should be stride fee = 5 DYDX reinvest = 90 DYDX.   

This assumes that 100 would be passed in so it can do the same computation as it did before, except the rebate was already taken out so instead it gets 95.  

Here is one way you could solve it: add an additional param flag which says if you are doing the rebate call or the stride fee call.  If doing the rebate call, the function would be as is currently.  If doing the stride fee call, you would scale the input value up as though the rebate hadn't already been removed and then do the computation the same way from that point on.  You would do this by computing the rebateFee, figuring out what percent of the total it is as rebateFeePct, then multiplying the current input by 1.0/(1.0 - rebateFeePct)  to get the original input.

So for example we compute the rebateFee is 4.75 DYDX of the incoming value of 95 DYDX, then we find that rebateFeePct = 5%, so we multiply the input of 95 DYDX by 1.0 / (1.0 - 0.05) to get 100 DYDX as the massaged input.  From that point on the computation is as normal so you would get rebateFee = 5 DYDX, stride fee = 5 DYDX and reinvest = 90 DYDX and you just ignore the rebate fee for this later call since it is intended for stride.

Reply:
In fact you don't have to compute the actual values you can find just the scale factor needed:

```
contributePercentage = ....

if strideFeeComputation {
    rebatePctOfTotal = strideFeePct * contributePercentage * rebateInfo.RebatePercentage
    scaleFactor = 1.0 / (1.0 - rebatePctOfTotal)
    rewardAmount *= scaleFactor
}

[Rest of computation as normal]
```

Reply:
took a stab at it in the latest commit